// http-exception-transformer/exceptions/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InternalServerException = exports.GoneException = exports.ConflictException = exports.RequestTimeOutException = exports.ProxyAuthenticationException = exports.NotAcceptableException = exports.MethodNotAllowedException = exports.NotFoundException = exports.ForbiddenException = exports.UnAuthorizedException = exports.BadRequestException = void 0;
var BadRequestException_gen_1 = __importDefault(require("./BadRequestException.gen"));
exports.BadRequestException = BadRequestException_gen_1.default;
var UnAuthorizedException_gen_1 = __importDefault(require("./UnAuthorizedException.gen"));
exports.UnAuthorizedException = UnAuthorizedException_gen_1.default;
var ForbiddenException_gen_1 = __importDefault(require("./ForbiddenException.gen"));
exports.ForbiddenException = ForbiddenException_gen_1.default;
var NotFoundException_gen_1 = __importDefault(require("./NotFoundException.gen"));
exports.NotFoundException = NotFoundException_gen_1.default;
var MethodNotAllowedException_gen_1 = __importDefault(require("./MethodNotAllowedException.gen"));
exports.MethodNotAllowedException = MethodNotAllowedException_gen_1.default;
var NotAcceptableException_gen_1 = __importDefault(require("./NotAcceptableException.gen"));
exports.NotAcceptableException = NotAcceptableException_gen_1.default;
var ProxyAuthenticationException_gen_1 = __importDefault(require("./ProxyAuthenticationException.gen"));
exports.ProxyAuthenticationException = ProxyAuthenticationException_gen_1.default;
var RequestTimeOutException_gen_1 = __importDefault(require("./RequestTimeOutException.gen"));
exports.RequestTimeOutException = RequestTimeOutException_gen_1.default;
var ConflictException_gen_1 = __importDefault(require("./ConflictException.gen"));
exports.ConflictException = ConflictException_gen_1.default;
var GoneException_gen_1 = __importDefault(require("./GoneException.gen"));
exports.GoneException = GoneException_gen_1.default;
var InternalServerException_gen_1 = __importDefault(require("./InternalServerException.gen"));
exports.InternalServerException = InternalServerException_gen_1.default;
