// http-exception-transformer/exceptions/MongoException.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoException = void 0;
var MongoException = /** @class */ (function (_super) {
    __extends(MongoException, _super);
    function MongoException(code, message, payload, callback) {
        if (callback === void 0) { callback = function () { return null; }; }
        var _this = _super.call(this, message) || this;
        _this.name = 'GenericMongoException';
        _this.message = message;
        _this.code = code;
        _this.payload = payload;
        _this.reference = 'https://github.com/stbrody/mongo/blob/master/docs/errors.md';
        _this.userCallback = callback;
        _this.userCallback();
        Object.setPrototypeOf(_this, MongoException.prototype);
        return _this;
    }
    MongoException.prototype.transformResponse = function () {
        return {
            code: this.code,
            error: true,
            message: this.message,
            payload: this.payload,
        };
    };
    return MongoException;
}(Error));
exports.MongoException = MongoException;
